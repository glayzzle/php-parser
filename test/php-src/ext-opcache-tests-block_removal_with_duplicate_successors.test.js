// eslint-disable prettier/prettier
const parser = require("../main");

describe("php-src tests", function () {
  // ext/opcache/tests/block_removal_with_duplicate_successors.phpt
  it("Removing a block that has duplicate successors", function () {
    expect(parser.parseCode("<?php\nfunction test($foo) {\n    $bar = 0;\n    if ($bar === 1 && $foo && PHP_SAPI !== 'cli') {\n        echo \"foo\\n\";\n    }\n    echo \"bar\\n\";\n}\ntest(1);\n?>")).toMatchSnapshot();
  });
});
