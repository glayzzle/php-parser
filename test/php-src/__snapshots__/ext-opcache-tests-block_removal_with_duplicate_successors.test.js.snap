// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`php-src tests Removing a block that has duplicate successors 1`] = `
Program {
  "children": Array [
    _Function {
      "arguments": Array [
        Parameter {
          "attrGroups": Array [],
          "byref": false,
          "flags": 0,
          "kind": "parameter",
          "name": Identifier {
            "kind": "identifier",
            "name": "foo",
          },
          "nullable": false,
          "readonly": false,
          "type": null,
          "value": null,
          "variadic": false,
        },
      ],
      "attrGroups": Array [],
      "body": Block {
        "children": Array [
          ExpressionStatement {
            "expression": Assign {
              "kind": "assign",
              "left": Variable {
                "curly": false,
                "kind": "variable",
                "name": "bar",
              },
              "operator": "=",
              "right": Number {
                "kind": "number",
                "value": "0",
              },
            },
            "kind": "expressionstatement",
          },
          If {
            "alternate": null,
            "body": Block {
              "children": Array [
                Echo {
                  "expressions": Array [
                    String {
                      "isDoubleQuote": true,
                      "kind": "string",
                      "raw": "\\"foo\\\\n\\"",
                      "unicode": false,
                      "value": "foo
",
                    },
                  ],
                  "kind": "echo",
                  "shortForm": false,
                },
              ],
              "kind": "block",
            },
            "kind": "if",
            "shortForm": false,
            "test": Bin {
              "kind": "bin",
              "left": Bin {
                "kind": "bin",
                "left": Bin {
                  "kind": "bin",
                  "left": Variable {
                    "curly": false,
                    "kind": "variable",
                    "name": "bar",
                  },
                  "right": Number {
                    "kind": "number",
                    "value": "1",
                  },
                  "type": "===",
                },
                "right": Variable {
                  "curly": false,
                  "kind": "variable",
                  "name": "foo",
                },
                "type": "&&",
              },
              "right": Bin {
                "kind": "bin",
                "left": Name {
                  "kind": "name",
                  "name": "PHP_SAPI",
                  "resolution": "uqn",
                },
                "right": String {
                  "isDoubleQuote": false,
                  "kind": "string",
                  "raw": "'cli'",
                  "unicode": false,
                  "value": "cli",
                },
                "type": "!==",
              },
              "type": "&&",
            },
          },
          Echo {
            "expressions": Array [
              String {
                "isDoubleQuote": true,
                "kind": "string",
                "raw": "\\"bar\\\\n\\"",
                "unicode": false,
                "value": "bar
",
              },
            ],
            "kind": "echo",
            "shortForm": false,
          },
        ],
        "kind": "block",
      },
      "byref": false,
      "kind": "function",
      "name": Identifier {
        "kind": "identifier",
        "name": "test",
      },
      "nullable": false,
      "type": null,
    },
    ExpressionStatement {
      "expression": Call {
        "arguments": Array [
          Number {
            "kind": "number",
            "value": "1",
          },
        ],
        "kind": "call",
        "what": Name {
          "kind": "name",
          "name": "test",
          "resolution": "uqn",
        },
      },
      "kind": "expressionstatement",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
